### PR TITLE
Actually issue bind requests

### PR DESCRIPTION
**Pull request by cpacejo**
This teaches the client adapter to issue a bind request to a dock.  It does so by queueing a request with the new adapter manager, which then performs the operation.

Missing is the reverse (dock issuing bind request to adapter).

Also missing is the node adapter internally generating its own tether IDs.

However this works between two adapters talking directly to each other.

Of course, the first two pings now get dropped.  (1st outbound to establish outbound flow, 2nd inbound to establish inbound flow.)

Also right now, the adapter manager does only one thing at a time.  In the future, maybe it will juggle multiple outstanding requests.

Sorry this PR is so involved.  I didn't have time to break it up into more pieces.  But you can look at the separate commits in the series to narrow down the scope into smaller bits.

Also:
* re-enables IPv6 in integration tests
* loosens criteria for capture test to pass
* makes DropGuard Send
* teaches sync req stuff (including Hello message) about link IDs
* moves sync req stuff (including Hello message) to mgmt module
* renames stream_id to tether_id when talking about tethers
* adds a "pending" state to the ALT
* copies in Sarah's packet debug code as a Debug trait (I needed it)
* deletes the hack to initially populate the ALT & DLT

### CODE REVIEW

File: adapter/ph/src/mgmt.rs [hunk 1]
```diff
@@ -167,20 +296,17 @@ pub async fn handle_bind_agent_address_request<'pktbuf>(
     _stream_id: zpr::StreamId, // ignored
     mut pkt: Packet<'pktbuf>,
 ) -> HandleMgmtResult<'pktbuf> {
-    let Some(hdr) = zdp::ZdpBindAgentAddressRequestHeader::ref_from_prefix(pkt.body()) else {
```
Comment (cpacejo) [inline adapter/ph/src/mgmt.rs:170]: This was actually bugged before!  Didn't strip off the header.  Fixed now.

File: adapter/ph/src/mgmt.rs [hunk 2]
```diff
@@ -85,6 +84,136 @@ pub async fn send_discard<'pktbuf>(asm: &'pktbuf Assembly<'pktbuf>, link_id: zpr
     send_non_flow_mgmt(asm, link_id, zdp::ZdpPacketType::Discard, pkt).await;
 }
 
+pub async fn send_hello_request<'a, 'pktbuf>(
```
Comment (cpacejo) [inline adapter/ph/src/mgmt.rs:90]: `send_hello_request` was only moved; slightly modified but not meaningfully so.

File: adapter/ph/src/mgmt.rs [hunk 3]
```diff
@@ -79,12 +79,145 @@ pub async fn send_report<'pktbuf>(
...
+        match self {
+            Self::SyncReqError(err) => err.fmt(f),
+            Self::BadStructure => write!(f, "bad structure"),
+            Self::BindAgentAddressError(msg) => f.write_str(&*msg),
+        }
+    }
+}
+
+/// send a Bind Agent Address Request and wait for the Response (RFC 6.5 § 6.3.11)
+pub async fn send_bind_agent_address_request<'a, 'pktbuf>(
```
Comment (cpacejo) [inline adapter/ph/src/mgmt.rs:139]: This does the legwork of formatting sending receiving & parsing the Bind request & response.

File: adapter/ph/src/fastpath.rs [hunk 4]
```diff
@@ -454,42 +457,51 @@ pub fn agent_output<'pktbuf>(asm: &Assembly<'pktbuf>, mut pkt: Packet<'pktbuf>)
...
+        Some(AltEntry::Pending) => {
+            // bind request pending; drop this packet
+            drop_and_count(asm, pkt, CounterType::DroppedAwaitingBind);
+        }
 
         None => {
-            // TODO: issue bind request!
-            drop_and_count(asm, pkt, CounterType::OtherError);
-            return;
+            // issue bind request
```
Comment (cpacejo) [inline adapter/ph/src/fastpath.rs:496]: Here we now issue a bind request, if the entry wasn't present in the ALT.

File: adapter/ph/src/assembly.rs [hunk 5]
```diff
@@ -234,27 +250,4 @@ impl<'pktbuf> Assembly<'pktbuf> {
             Err(_) => return Err(err_type),
         }
     }
-
-    pub async fn send_hello_req(&self) {
```
Comment (cpacejo) [inline adapter/ph/src/assembly.rs:238]: Moved to mgmt.

File: adapter/ph/src/adapter_manager_worker.rs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+pub fn launch<'pktbuf>(
+    asm: impl std::ops::Deref<Target = Assembly<'pktbuf>> + Send + Sync + 'pktbuf,
+    mut queue: mpsc::Receiver<AdapterManagerMessage<'pktbuf>>,
+) -> impl Future<Output = ()> + Send + 'pktbuf {
+    async move { worker(&*asm, &mut queue).await }
+}
+
+// RFC 6.5 § 6.3.11
+async fn do_request_tether_id<'pktbuf>(asm: &Assembly<'pktbuf>, pkt: Packet<'pktbuf>) {
```
Comment (cpacejo) [inline adapter/ph/src/adapter_manager_worker.rs:36]: This implements the semantics of requesting a tether ID -- issues the bind request and sticks the result into the ALT.

**Review by svjaffer — APPROVED** [review]

**Comment by cpacejo** [comment]
Completes #363.

**Pull request merged**